geyser-plugin-manager/src/geyser_plugin_service.rs

```diff
@@ -208,4 +221,384 @@ impl GeyserPluginService {
         self.plugin_manager.write().unwrap().unload();
         Ok(())
     }
+
+    pub fn plugin_manager_join_handle(&self) -> Arc<JoinHandle<()>> {
+        // This is only called after the initialization of a solana-test-validator or a solana-validator
+        // which has a geyser plugin service
+        Arc::clone(self.rpc_handler_thread.as_ref().unwrap())
+    }
+
+    pub fn list_plugins_rpc(plugin_manager: &GeyserPluginManager) -> JsonRpcResult<Vec<String>> {
+        Ok(plugin_manager
+            .plugins
+            .iter()
+            .map(|p| p.name().to_owned())
+            .collect())
+    }
+
+    pub fn load_plugin_rpc(
+        plugin_manager: &mut GeyserPluginManager,
+        config_file: &str,
+    ) -> JsonRpcResult<String> {
+        // Load plugin
+        let mut new_plugin: Box<dyn GeyserPlugin> = load_plugin_from_config(Path::new(config_file))
+            .map_err(|err| jsonrpc_core::Error {
+                code: ErrorCode::InternalError,
+                message: err.to_string(),
+                data: None,
+            })?;
+
+        // Then see if a plugin with this name already exists. If so, abort
+        if plugin_manager
```

Comment:
Why are we loading first and check later? Would not you find yourself?

Reply:
To get the name of the plugin, you need to call the `name` method which can only be called once you load it.